walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,151 @@
+    boolean overlapped = false;
+    Event nextEvent = null;
+
+    //Iterates through sorted important events to find the meeting times
+    for (Iterator iterator = set.iterator(); iterator.hasNext();) {
+        
+        Map.Entry mentry = (Map.Entry)iterator.next();
+        
+        //currentEvent is the event the loop is currently on
+        Event currentEvent = (Event)mentry.getValue(); 
+        
+        //nextEvent is the event that comes after the currentEvent in the TreeMap
+        if (iterator.hasNext()) { 
+            nextEvent = (Event)eventsList.higherEntry((int)mentry.getKey()).getValue();
+        }
+        
+        //in case two events overlap, and one contains the other, and there are no other events after those
+        if (contained && !iterator.hasNext()) {
+            //create new TimeRange
+            TimeRange newRange = TimeRange.fromStartEnd(limitTime, TimeRange.END_OF_DAY, true);
+            //if the duration of the new TimeRange is greater or equal to the duration of the meeting requested
+            //then this is a TimeRange we can use for the meeting to occur
+            if (newRange.duration() >= request.getDuration()) {
+                finalTimes.add(newRange);
+            }
+        }
+        //in case two events overlap but one does not contain the other, and there are no other events after those
+        else if (overlapped && !iterator.hasNext()) {
+            //create new TimeRange
+            TimeRange newRange = TimeRange.fromStartEnd(limitTime, TimeRange.END_OF_DAY, true);
+            //if the duration of the new TimeRange is greater or equal to the duration of the meeting requested
+            //then this is a TimeRange we can use for the meeting to occur
+            if (newRange.duration() >= request.getDuration()) {
+                finalTimes.add(newRange);
+            }
+        }
+        else {
+            contained = false;
+            overlapped = false;
+            //create new TimeRange
+            TimeRange newRange = TimeRange.fromStartEnd(limitTime, currentEvent.getWhen().start(), false);
+            //if the duration of the new TimeRange is greater or equal to the duration of the meeting requested
+            //then this is a TimeRange we can use for the meeting to occur
+            if (newRange.duration() >= request.getDuration()) {
+                finalTimes.add(newRange);
+            }
+            //if there is a next event
+            if (iterator.hasNext()) { 
+                //if current event does not overlap next event
+                if (!currentEvent.getWhen().overlaps(nextEvent.getWhen())) { 
+                    //limitTime becomes the currentEvent's end time
+                    limitTime = currentEvent.getWhen().end(); 
+                }
+                //if events overlap
+                else { 
+                    //if currentEvent contains nextEvent 
+                    if (currentEvent.getWhen().contains(nextEvent.getWhen())) {
```

Comment:
Not sure if this will ever be called since no two entries in your code would have the same start time since all of the keys in your map must be unique